pkg/serving/v1/gitops.go

```
@@ -0,0 +1,242 @@
+// Copyright 2020 The Knative Authors
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+
+	"knative.dev/client/pkg/wait"
+
+	"k8s.io/apimachinery/pkg/watch"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+const (
+	operationNotSuportedError = "this operation is not supported in gitops mode"
+	ksvcKind                  = "ksvc"
+)
+
+// knServingGitOpsClient - kn service client
+// to work on a local repo instead of a remote cluster
+type knServingGitOpsClient struct {
+	dir       string
+	namespace string
+	printer   printers.ResourcePrinter
+}
+
+// NewKnServingGitOpsClient returns an instance of the
+// kn service gitops client
+func NewKnServingGitOpsClient(namespace, dir string) KnServingClient {
+	yamlPrinter, _ := genericclioptions.NewJSONYamlPrintFlags().ToPrinter("yaml")
```

Ignoring errors... I would rather propagate or log at minimum